packages/libs/deer/src/error/mod.rs

```diff
@@ -83,6 +104,78 @@ impl Id {
     }
 }
 
+// TODO: most likely (in 0.2) we want to actually have a proper schema
+pub struct Schema {
+    ty: String,
+    other: BTreeMap<String, Box<dyn erased_serde::Serialize + Send + Sync>>,
+}
+
+impl Schema {
+    #[must_use]
+    pub fn new(ty: impl Into<String>) -> Self {
+        Self {
+            ty: ty.into(),
+            other: BTreeMap::new(),
+        }
+    }
+
+    pub(crate) fn ty(&self) -> &str {
+        &self.ty
+    }
+
+    #[must_use]
+    pub fn with(
+        mut self,
+        key: impl Into<String>,
+        value: impl erased_serde::Serialize + Send + Sync + 'static,
+    ) -> Self {
+        self.other.insert(key.into(), Box::new(value));
+
+        self
+    }
+
+    pub fn set(
+        &mut self,
+        key: impl Into<String>,
+        value: impl erased_serde::Serialize + Send + Sync + 'static,
+    ) -> &mut Self {
+        self.other.insert(key.into(), Box::new(value));
+
+        self
+    }
+}
+
+impl Serialize for Schema {
+    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(Some(self.other.len() + 1))?;
+
+        map.serialize_entry("type", &self.ty)?;
+
+        for (key, value) in &self.other {
+            map.serialize_entry(key, value)?;
+        }
+
+        map.end()
+    }
+}
+
+pub(crate) fn fold_field<'a>(it: impl Iterator<Item = &'a str>) -> String {
```

Comment:
Did you consider Instead of folding using the `Formatter` directly in the implementation by passing it from the message function?

Reply:
nice catch! fixed in [c42234e](https://github.com/hashintel/hash/pull/1375/commits/c42234ed02fb7f2d0523c22e238926ed4d1f264b)